### PR TITLE
DHistogram.NASplitDir 

### PR DESCRIPTION
**Pull request by st-pasha**
In DHistogram.NASplitDir assign an explicit integer value to each enum constant, so that it can be serialized/deserialized more reliably (including across packages).

---
This change is [Reviewable](https://reviewable.io/reviews/h2oai/h2o-3/127)

### CODE REVIEW

File: h2o-algos/src/main/java/hex/tree/CompressedTree.java [hunk 1]
```diff
@@ -20,15 +20,26 @@
...
+    _nclass = nclass;
+    _seed = seed;
   }
 
-  /** Highly efficient (critical path) tree scoring */
+  /**
+   * Highly efficient (critical path) tree scoring
+   *
+   * WARNING: If you making any changes to this code, you should synchronize them with the class
+   * `hex.genmodel.algos.DrfRawModel` in package `h2o_genmodel`.
```
Comment (arnocandel) [inline h2o-algos/src/main/java/hex/tree/CompressedTree.java]: Not just DRF, but also GBM

**Comment by arnocandel** [comment]
Looks good.

**Comment by mmalohlava** [comment]
:+1:

**Comment by arnocandel** [comment]
will merge once the comment is updated.

**Pull request merged**